runtime/compiler/env/j9method.cpp

```diff
@@ -5313,6 +5313,34 @@ TR_ResolvedJ9Method::isJITInternalNative()
    return isNative() && !isJNINative() && !isInterpreted();
    }
 
+void
+TR_ResolvedJ9Method::getFaninInfo(uint32_t *count, uint32_t *weight, uint32_t *otherBucketWeight)
+   {
+   TR_IProfiler *profiler = _fe->getIProfiler();
+
+   if (!profiler)
+      {
+      return;
+      }
+
+   TR_OpaqueMethodBlock *method = getPersistentIdentifier();
```

Comment:
My suggestion would be to create a getFaninInfo at the IProfiler as well and call it here. Thus we can have one virtual call instead of two.
`profiler->getNumberofCallersAndTotalWeight()` and `profiler->getOtherBucketWeight()` can be eliminated then.